Dispatcher/UI/Auth/AuthViewController.swift

```diff
@@ -7,57 +7,196 @@
+    @IBOutlet var contentView: UIView!
+    @IBOutlet weak var headerView: HeaderView!
+    @IBOutlet weak var authView: UIView!
+    
+    @IBOutlet weak var emailTextField: TextFieldWithValidation!
+    @IBOutlet weak var emailErrorLabel: UILabel!
+    
+    @IBOutlet weak var passwordTextField: TextFieldWithValidation!
+    @IBOutlet weak var passwordErrorLabel: UILabel!
+    
+    @IBOutlet weak var reEnterPasswordTextField: TextFieldWithValidation!
+    @IBOutlet weak var reEnterPasswordErrorLabel: UILabel!
+    
+    @IBOutlet weak var authModeLabel: UILabel!
+    @IBOutlet weak var primaryButton: DispatcherAppButton!
+    @IBOutlet weak var secondaryButton: DispatcherAppButton!
+    
+    var authViewModel = AuthViewModel.shared
+    var authMode: AuthMode = .Signup
+    var validatedEmail: String = ""
+    var validatedPassword: String = ""
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        let screenHeight = UIScreen.main.bounds.height
-        let screenWidth = UIScreen.main.bounds.width
-                
-        logoHeader = HeaderView(frame: CGRect(x: 0, y: 0, width: screenWidth, height: screenHeight * 0.45), headerType: .logoHeader)
-        self.view.addSubview(logoHeader ?? UIView())
+        headerView.initView(headerType: .logoHeader)
+        emailTextField.initTextField(type: .email)
+        passwordTextField.initTextField(type: .password)
+        reEnterPasswordTextField.initTextField(type: .reEnterPssword)
         
-        signupOrLoginView = SignupOrLoginView(frame: CGRect(x: 0, y: 0, width: screenWidth, height: screenHeight * 0.55))
-        self.view.addSubview(signupOrLoginView ?? UIView())
-        signupOrLoginView?.delegate = self
+        emailTextField?.validationDelegate = self
+        passwordTextField?.validationDelegate = self
+        reEnterPasswordTextField?.validationDelegate = self
         
-        setViewsConstraints()
+        setSignupView()
     }
         
-    private func setViewsConstraints() {
-        if let logoHeader {
-            logoHeader.translatesAutoresizingMaskIntoConstraints = false
-            NSLayoutConstraint.activate([
-                logoHeader.topAnchor.constraint(equalTo: self.view.topAnchor),
-                logoHeader.leadingAnchor.constraint(equalTo: self.view.leadingAnchor),
-                logoHeader.trailingAnchor.constraint(equalTo: self.view.trailingAnchor),
-                logoHeader.heightAnchor.constraint(equalTo: self.view.heightAnchor, multiplier: 0.45)
-            ])
+        private func setLoginView() {
+            primaryButton.initDispatcherButton(type: .primary, title: AuthMode.Login.rawValue.uppercased() , icon: UIImage(named: "Arrow - Right"), iconPosition: .end, cornerRadius: 10)
+            
+            secondaryButton.initDispatcherButton(type: .secondary, title: AuthMode.Signup.rawValue.uppercased(), cornerRadius: 10)
+            
+            reEnterPasswordTextField.isHidden = true
```

Comment:
Maybe you can create a toggleReEnterPasswordVisibility function for this and use it instead of declaring manually both in this line and in line 64-65